runner/operator/access/v1_0_0/merge_fastqs/__init__.py

```diff
@@ -0,0 +1,111 @@
+import os
+from collections import defaultdict
+from itertools import groupby
+from runner.operator.operator import Operator
+from runner.serializers import APIRunCreateSerializer
+from file_system.repository.file_repository import FileRepository
+
+import json
+from jinja2 import Template
+
+WORKDIR = os.path.dirname(os.path.abspath(__file__))
+
+
+METADATA_OUTPUT_FIELDS = [
+    'barcodeId',
+    'sampleName',
+    'investigatorSampleId',
+    'patientId',
+    'tumorOrNormal',
+    'sampleOrigin',
+    'dnaInputNg',
+    'captureInputNg',
+    'baitSet',
+    'sex',
+    'barcodeIndex',
+    'libraryVolume',
+    'captureName',
+    'libraryConcentrationNgul'
+]
+def construct_inputs(samples):
+    with open(os.path.join(WORKDIR, 'input_template.json.jinja2')) as file:
+        template = Template(file.read())
+
+
+    samples = list(group_by_sample_id(samples).values())
+
+    inputs = list()
+    for sample_files in samples:
+        fastqs = group_by_fastq(sample_files)
+
+        fastq1s = [{
+            "class": "File",
+            "location": "juno://" + fastq["path"]
+        } for fastq in fastqs["R1"]]
+
+        fastq2s = [{
+            "class": "File",
+            "location": "juno://" + fastq["path"]
+        } for fastq in fastqs["R2"]]
+
+        input_file = template.render(
+            fastq1_files=json.dumps(fastq1s),
+            fastq2_files=json.dumps(fastq2s),
+        )
+
+        inputs.append((json.loads(input_file), sample_files[0]["metadata"]))
+
+    return inputs
+
+class AccessLegacyFastqMergeOperator(Operator):
+    def get_jobs(self):
+        files = FileRepository.filter(queryset=self.files,
+                                      metadata={'requestId': self.request_id,
+                                                'igocomplete': True})
+        data = [
+            {
+                "id": f.file.id,
+                "path": f.file.path,
+                "file_name": f.file.file_name,
+                "metadata": f.metadata
+            } for f in files
+        ]
+
+        inputs = construct_inputs(data)
+
+        number_of_inputs = len(inputs)
+
+        return [
+            (
+                APIRunCreateSerializer(
+                    data={
+                        'name': "LEGACY FASTQ Merge: %s, %i of %i" % (self.request_id, i + 1, number_of_inputs),
+                        'app': self.get_pipeline_id(),
+                        'output_metadata': {key: metadata[key] for key in METADATA_OUTPUT_FIELDS if
+                                            key in metadata},
```

Comment:
I'm not sure if you were planning to remove the `igoComplete: true` filter from AccessLegacyOperator, but if not then we need to pass that field here as well

Reply:
I think we can remove it. Since we're using it here, all the files are passed are `igoComplete`. It just means we have to use `self.run_ids` for the BAM generation, which I think is for the best anyways. But  let me know if you disagree.